src/masternodes/rpc_icxorderbook.cpp

```diff
@@ -0,0 +1,1497 @@
+    CWallet* const pwallet = GetWallet(request);
+
+    RPCHelpMan{"icx_closeorder",
+                "\nCloses (and submits to local node and network) order transaction.\n" +
+                HelpRequiringPassphrase(pwallet) + "\n",
+                {
+                    {"orderTx", RPCArg::Type::STR, RPCArg::Optional::NO, "txid of maker order"},
+                    {"inputs", RPCArg::Type::ARR, RPCArg::Optional::OMITTED_NAMED_ARG,
+                        "A json array of json objects",
+                        {
+                            {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                                {
+                                    {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                                    {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                },
+                            },
+                        },
+                    },
+                },
+                
+                RPCResult{
+                        "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+                },
+                RPCExamples{
+                        HelpExampleCli("closeorder", "'{\"orderTx\":\"<txid>>\"}'")
+                },
+     }.Check(request);
+
+    if (pwallet->chain().isInitialBlockDownload())
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Cannot close order while still in Initial Block Download");
+
+    pwallet->BlockUntilSyncedToCurrentChain();
+    LockedCoinsScopedGuard lcGuard(pwallet);
+
+    RPCTypeCheck(request.params, {UniValue::VSTR}, false);
+    if (request.params[0].isNull())
+        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                           "Invalid parameters, arguments 1 must be non-null and expected as \"orderTx\"}");
+
+    UniValue const & txInputs = request.params[1];
+
+    CICXCloseOrder closeorder;
+    closeorder.orderTx = uint256S(request.params[0].getValStr());
+
+    int targetHeight;
+    {
+        LOCK(cs_main);
+        auto order = pcustomcsview->GetICXOrderByCreationTx(closeorder.orderTx);
+        if (!order)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "orderTx (" + closeorder.orderTx.GetHex() + ") does not exist");
+        if (!order->closeTx.IsNull()) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER,"orderTx (" + closeorder.orderTx.GetHex() + " is already closed!");
+        }
+
+        targetHeight = ::ChainActive().Height() + 1;
+    }
+
+    CDataStream metadata(DfTxMarker, SER_NETWORK, PROTOCOL_VERSION);
+    metadata << static_cast<unsigned char>(CustomTxType::ICXCloseOrder)
+             << closeorder;
+
+    CScript scriptMeta;
+    scriptMeta << OP_RETURN << ToByteVector(metadata);
+
+    const auto txVersion = GetTransactionVersion(targetHeight);
+    CMutableTransaction rawTx(txVersion);
+
+    CTransactionRef optAuthTx;
+    std::set<CScript> auths;
```

Comment:
No auth set, how do we authorise a close order TX? Do we need to set an auth on open order and make a corresponding auth here.